src/common/offline_transformations/src/compress_quantize_weigths.cpp

```diff
@@ -220,8 +222,93 @@ ov::pass::CompressQuantizeWeights::CompressQuantizeWeights() {
+    auto weights_convert_pattern = pattern::wrap_type<op::v0::Convert>({weights_const_pattern});
+    OutputVector weights_options{weights_const_pattern, weights_convert_pattern};
+    auto weights_pattern = std::make_shared<pattern::op::Or>(weights_options);
+    auto fake_convert_pattern = pattern::wrap_type<op::v13::FakeConvert>(
+        {weights_pattern, pattern::wrap_type<op::v0::Constant>(), pattern::wrap_type<op::v0::Constant>()});
+    auto fake_convert_pattern2 =
+        pattern::wrap_type<op::v13::FakeConvert>({weights_pattern, pattern::wrap_type<op::v0::Constant>()});
+    auto root = std::make_shared<pattern::op::Or>(OutputVector{fake_convert_pattern, fake_convert_pattern2});
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_map();
+        const auto fake_convert = ov::as_type_ptr<op::v13::FakeConvert>(m.get_match_root());
+        auto weights = pattern_map.at(weights_const_pattern);
+
+        NodeVector from{weights, fake_convert, fake_convert->get_input_node_shared_ptr(1)};
+        NodeRegistry node_registry;
+
+        if (weights->get_output_element_type(0) != fake_convert->get_input_element_type(0)) {
+            weights = std::make_shared<op::v0::Convert>(weights, fake_convert->get_input_element_type(0));
+        }
+        const auto scale = fake_convert->input_value(1);
+        weights = std::make_shared<op::v1::Multiply>(weights, scale);
+        const auto shift = get_fake_convert_shift(fake_convert);
+        if (shift) {
+            from.push_back(shift);
+            weights = std::make_shared<op::v1::Subtract>(weights, shift);
+        }
+        const auto destination_type = element::Type(fake_convert->get_destination_type());
+        const auto weights_convert = std::make_shared<op::v0::Convert>(weights, destination_type);
+        auto compressed_weights = ov::util::constantfold_subgraph(weights_convert);
+        if (!compressed_weights) {
+            return false;
+        }
+        node_registry.add(compressed_weights);
+
+        const auto convert =
+            node_registry.make<op::v0::Convert>(compressed_weights, fake_convert->get_input_element_type(0));
+        const auto inv_scale = ov::util::constantfold_subgraph(
+            std::make_shared<op::v1::Power>(scale,
+                                            op::v0::Constant::create(scale.get_element_type(), Shape{}, {-1.0f})));
+        if (!inv_scale)
+            return false;
+        node_registry.add(inv_scale);
+        std::shared_ptr<op::v1::Multiply> multiply;
+        if (shift) {
+            // TODO: check if shift can be fused to weights and eliminate it
+            const auto neg_shift = ov::util::constantfold_subgraph(std::make_shared<op::v0::Negative>(shift));
+            if (!neg_shift)
+                return false;
+            node_registry.add(neg_shift);
+            const auto subtract = node_registry.make<op::v1::Subtract>(convert, neg_shift);
```

Comment:
Is there any advantage of usage
`Subtract(..., Negative(shift))` instead of `Add`?
In other words:
`data - (-shift)` instead of `data + shift`

Reply:
Convert->Subtract->Multiply is consistent with dequantization subgraph that LPT supports.